Kernel/Storage/ATA/GenericIDE/Controller.h

```diff
@@ -31,6 +31,6 @@ class IDEController : public ATAController {
     IDEController();
 
     LockRefPtr<StorageDevice> device_by_channel_and_position(u32 index) const;
-    Vector<NonnullLockRefPtr<IDEChannel>> m_channels;
+    Array<RefPtr<IDEChannel>, 2> m_channels;
```

Comment:
Can this also be `NonnullRefPtr`?

Reply:
No, because we are only creating the IDE channel object after the construction point.